static/js/components/ProfileImageUploader.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import React from 'react';
+import Dialog from 'material-ui/Dialog';
+import R from 'ramda';
+import Dropzone from 'react-dropzone';
+import Cropper from 'react-cropper';
+import Button from 'react-mdl/lib/Button';
+import type { ImageUploadState } from '../reducers/image_upload';
+
+const onDrop = R.curry((startPhotoEdit, files) => startPhotoEdit(...files));
+
+const dropZone = onDrop => (
+  <Dropzone
+    onDrop={onDrop}
+    className="photo-active-item photo-dropzone"
+    activeClassName="photo-active-item photo-dropzone active"
+  >
+    <div>
+      Drag a photo here or click to select a photo.
+    </div>
+  </Dropzone>
+);
+
+class CropperWrapper extends React.Component {
+  props: {
+    updatePhotoEdit:  (b: Blob) => void,
+    photo:            Object,
+  };
+
+  cropperHelper = () => {
+    const { updatePhotoEdit } = this.props;
+    this.refs.cropper.getCroppedCanvas().toBlob(blob => updatePhotoEdit(blob));
+  };
+
+  render () {
+    const { photo } = this.props;
+    return <Cropper
+      ref='cropper'
+      className="photo-active-item cropper"
+      src={photo.preview}
+      aspectRatio={ 1 / 1 }
+      guides={false}
+      cropend={this.cropperHelper}
+      ready={this.cropperHelper}
+    />;
+  }
+}
+
+type ImageUploadProps = {
+  photoDialogOpen:      boolean;
+  setDialogVisibility:  (b: boolean) => void;
+  startPhotoEdit:       (p: File) => void;
+  clearPhotoEdit:       () => void;
+  imageUpload:          ImageUploadState;
+  updateUserPhoto:      (i: string) => Promise<*>;
```

Comment:
Can we be more specific with the type of `Promise` or is this difficult in Flow?